tracer/src/Datadog.Trace/Ci/Agent/MessagePack/CIEventMessagePackFormatter.cs

```diff
@@ -0,0 +1,199 @@
+// <copyright file="CIEventMessagePackFormatter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Ci.Agent.Payloads;
+using Datadog.Trace.Ci.Tags;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.PlatformHelpers;
+using Datadog.Trace.Vendors.MessagePack;
+
+namespace Datadog.Trace.Ci.Agent.MessagePack
+{
+    internal class CIEventMessagePackFormatter : EventMessagePackFormatter<EventsPayload>
+    {
+        // .
+        private readonly byte[] _metadataBytes = StringEncoding.UTF8.GetBytes("metadata");
+        private readonly byte[] _containerIdBytes = StringEncoding.UTF8.GetBytes("container_id");
+        private readonly byte[] _containerIdValueBytes;
```

Comment:
You might find it helpful to add `#nullable enable` to this file, given that _some_ of these `byte[]` can be `null`, and some can't